fe/fe-core/src/main/java/org/apache/doris/nereids/rules/exploration/join/InnerJoinLAsscomProject.java

```diff
@@ -137,17 +127,43 @@ public Rule build() {
                     JoinUtils.addSlotsUsedByOn(bUsedSlots, newRightProjects);
                     JoinUtils.addSlotsUsedByOn(aUsedSlots, newLeftProjects);
 
-                    if (!newLeftProjects.isEmpty()) {
-                        newLeftProjects.addAll(cOutputSet);
-                    }
-
                     /* ********** new Plan ********** */
-                    LogicalJoin<GroupPlan, GroupPlan> newBottomJoin = new LogicalJoin<>(topJoin.getJoinType(),
-                            newBottomHashConjuncts, newBottomOtherConjuncts, JoinHint.NONE,
-                            a, c, bottomJoin.getJoinReorderContext());
+                    LogicalJoin newBottomJoin;
+                    if (needNewProjectChildForA) {
+                        /*
+                        *        topJoin                   newTopJoin
+                        *        /     \                   /        \
+                        *    project    C          newLeftProject newRightProject
+                        *      /            ──►          /            \
+                        * bottomJoin                newBottomJoin      B
+                        *    /   \                     /   \
+                        *   A     B                   A     C
+                        *                            /
+                        *                  needNewProjectChildForA
+                        */
+                        // create a new project node as A's child
+                        newBottomJoin = new LogicalJoin<>(topJoin.getJoinType(),
+                                newBottomHashConjuncts, newBottomOtherConjuncts,
+                                JoinHint.NONE, JoinReorderUtils.projectOrSelf(newLeftProjects, a), c,
+                                bottomJoin.getJoinReorderContext());
+                    } else {
+                        // replace the join conjuncts
+                        newBottomHashConjuncts = JoinUtils.replaceJoinConjuncts(
+                                newBottomHashConjuncts, replaceMapForNewBottomJoin);
+                        newBottomOtherConjuncts = JoinUtils.replaceJoinConjuncts(
+                                newBottomOtherConjuncts, replaceMapForNewBottomJoin);
+                        newBottomJoin = new LogicalJoin<>(topJoin.getJoinType(),
+                                newBottomHashConjuncts, newBottomOtherConjuncts,
+                                JoinHint.NONE, a, c, bottomJoin.getJoinReorderContext());
+                    }
                     newBottomJoin.getJoinReorderContext().setHasLAsscom(false);
                     newBottomJoin.getJoinReorderContext().setHasCommute(false);
 
+                    // new left project should contain all output slots from C
+                    if (!newLeftProjects.isEmpty()) {
+                        newLeftProjects.addAll(cOutputSet);
+                    }
```

Comment:
has bugs? newLeftProject as done under A if needNewProjectChildForA, do it again may lead to some exprs contains slots not from its child